storage/cloud-client/notification_polling.py

```diff
@@ -0,0 +1,169 @@
+
+
+def Summarize(message):
+    # [START parse_message]
+    data = message.data
+    attributes = message.attributes
+
+    # The kind of event that just happened. Example: OBJECT_FINALIZE
+    event_type = attributes['eventType']
+    event_description = ACTION_STRINGS.get(
+        event_type, 'Unknown Event %s' % event_type)
+
+    # ID of the bucket. Example: "my_photos"
+    bucket_id = attributes['bucketId']
+    # The ID of the affected objet. Example: "image.jpeg"
+    object_id = attributes['objectId']
+    # Generation of the object. Example: 1234567
+    generation = attributes['objectGeneration']
+    # Format of the attached payload. Example: JSON_API_V1
+    payload_format = attributes['payloadFormat']
+    description = (
+        '{summary} - {bucket_id}/{object_id}\n'
+        '\tGeneration: {generation}\n').format(
+            summary=event_description,
+            bucket_id=bucket_id,
+            object_id=object_id,
+            generation=generation)
+
+    if payload_format == 'JSON_API_V1':
+        # The payload is the JSON API object resource.
+        object_metadata = json.loads(data)
+        size = object_metadata['size']
+        content_type = object_metadata['contentType']
+        metageneration = object_metadata['metageneration']
+        description += (
+            '\tContent type: {content_type}\n'
+            '\tSize: {object_size}\n'
+            '\tMetageneration: {metageneration}\n').format(
+                content_type=content_type,
+                object_size=size,
+                metageneration=metageneration)
+    elif payload_format == 'NONE':
+        # There is no payload.
+        pass
+    else:
+        print('Unknown payload format %s', payload_format)
+    return description
+    # [END parse_message]
+
+
+def GetDedupString(message):
+    """Returns a string unique to this specific event."""
+    # GCS will publish each notification at least once to a Cloud Pub/Sub
+    # topic, and Cloud Pub/Sub will deliver each message to each
+    # subscription at least once. We must be able to safely handle
+    # occasionally receiving duplicate messages.
+    metageneration = 'unknown'
+    if message.attributes['payloadFormat'] == 'JSON_API_V1':
+        metageneration = json.loads(message.data)['metageneration']
+    return DUPLICATION_KEY.format(
+        event_type=message.attributes['eventType'],
+        resource=message.attributes['resource'],
+        metageneration=metageneration)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description=__doc__)
+    parser.add_argument(
+        '--project',
+        help='The project of the subscription, if not in your default project')
+    parser.add_argument('subscription',
+                        help='The ID of the Pub/Sub subscription')
+    args = parser.parse_args()
+
+    subscription_id = args.subscription
+    client = pubsub.Client(project=args.project)
```

Comment:
Put this line and everything below into a "poll_notifications" function, please.